### PR TITLE
EVM Client Improvements and Tests

### PR DESCRIPTION
**Pull request by aludvik**
This PR does some reorganization, separates out the client from the cli, adds integration tests for the burrow-evm transaction family, and fixes a few other issues.

### CODE REVIEW

File: families/burrow_evm/src/sawtooth_burrow_evm/cli/exec.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+func (e *Exec) Name() string {
+	return "exec"
+}
+
+func (e *Exec) Register(p *flags.Parser) error {
+	_, err := p.AddCommand("exec", "Call an existing account", "", e)
+	return err
+}
+
+func (e *Exec) Do() error {
```
Comment (peterschwarz) [inline families/burrow_evm/src/sawtooth_burrow_evm/cli/exec.go]: `Do` is not a great name for this.   `Execute`,  or `Run` are far better names here.

File: integration/sawtooth_integration/docker/burrow-evm-smoke.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  rest_api:
+    image: sawtooth-rest_api:$ISOLATION_ID
+    volumes:
+      - $SAWTOOTH_CORE:/project/sawtooth-core
+    expose:
+      - 40000
+      - 8080
+    depends_on:
+      - validator
+    command: rest_api -v --stream-url tcp://validator:40000 --host rest_api
```
Comment (agunde406) [inline integration/sawtooth_integration/docker/burrow-evm-smoke.yaml]: This needs to be updated to match the new rest_api command line options.

File: integration/sawtooth_integration/docker/burrow-evm-smoke.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      - $SAWTOOTH_CORE:/project/sawtooth-core
+    expose:
+      - 40000
+      - 8800
+    # start the validator with an empty genesis batch
+    command: "bash -c \"\
+        sawtooth admin keygen && \
+        sawtooth admin genesis && \
+        validator -v \
+            --public-uri tcp://validator:8800 \
```
Comment (agunde406) [inline integration/sawtooth_integration/docker/burrow-evm-smoke.yaml]: This needs to be updated to match the new validator command line options.

**Comment by aludvik** [comment]
Changed Do() -> Run()

**Pull request merged**